### ISSUE TITLE
[feature request] make the preprocessing more generic

### ISSUE THREAD

**Issue by ad34** [jisungk/deepjazz#7]
I'm trying this great work to generate music from other midi files but there is some kinda "magic" parts in the preprocess.py file for me.

some code parts use magic numbers like

melody_stream = midi_data[5] 
partIndices = [..]

other require strong knowledge of the example midi file & music21 library.

I manage to achieve the __parse_midi function to return values without exception but got an error later in __get_abstract_grammars (return empty list)

after change I changed 4 more thing and finally the training started but got a value error in __generate_grammar after a few iterations.

Basically I would like to make the generator to work with a preprocessed midi file with single channel .

**Comment by Echo-Ji** [comment]
I got the same problem when i try use my own MIDI file.
I think there is some knowledge about music waiting to learn.:）

**Comment by cvantass** [comment]
Pretty late to the party here, and maybe this issue has already been solved, but did either of you figure this out? I'm having the exact same problem here. I took one of my own midi files that is just one single part, but it won't go through the preprocessing. If either of you still have the code around, what did you end up doing about this? Thanks and sorry for being a few years late...

**Comment by kairanaquazi** [comment]
im even later and now the code doesn't even work on the original file :(
